warp10/src/main/java/io/warp10/script/functions/FETCH.java

```diff
@@ -1485,4 +1537,154 @@ public static Long getTimestamp(Object timestampRepresentation, String timestamp
 
     return timestamp;
   }
+
+  public static Map<String,Object> parseTokenAttributes(ReadToken token, Map<String,Object> params) throws WarpScriptException {
+    Map<String,Object> attributes = new HashMap<String,Object>();
+
+    Map<String,String> tokenattr = token.getAttributes();
+
+    if (null == tokenattr) {
+      return attributes;
+    }
+
+    long now = TimeSource.getTime();
+
+    Instant ref = new Instant(now / Constants.TIME_UNITS_PER_MS);
+
+    String value = tokenattr.get(Constants.TOKEN_ATTR_END);
+
+    Long end = null;
+
+    if (null != value) {
+      value = value.trim();
+
+      if (value.startsWith("P")) {
+        long offset = DURATION.parseDuration(ref, value, true, true);
+        end = now + offset;
+      } else {
+        end = TOTIMESTAMP.parseTimestamp(value);
+      }
+      attributes.put(PARAM_END, end);
+    }
+
+    value = tokenattr.get(Constants.TOKEN_ATTR_START);
+    if (null != value) {
+      value = value.trim();
+      long start;
+
+      if (value.startsWith("P")) {
+        long offset = DURATION.parseDuration(ref, value, true, true);
+        if (null == end) {
+          if (null != params && params.get(PARAM_END) instanceof Long) {
+            start = ((Long) params.get(PARAM_END)).longValue() + offset;
+          } else {
+            start = now + offset;
+          }
+        } else {
+          start = end + offset;
+        }
+      } else {
+        start = TOTIMESTAMP.parseTimestamp(value);
+      }
+      attributes.put(PARAM_START, start);
+    }
+
+    value = tokenattr.get(Constants.TOKEN_ATTR_QUIETAFTER);
+    if (null != value) {
+      value = value.trim();
+      long quietafter;
+
+      if (value.startsWith("P")) {
+        long offset = DURATION.parseDuration(ref, value, true, true);
+        if (null == end) {
+          if (null != params && params.get(PARAM_END) instanceof Long) {
+            quietafter = ((Long) params.get(PARAM_END)).longValue() + offset;
+          } else {
+            quietafter = now + offset;
+          }
+        } else {
+          quietafter = end + offset;
+        }
+        quietafter = now + offset;
```

Comment:
If you want to keep the logic above this line, remove this line as it overrides the value set above in the conditional blocks.